File: tauri-app/src/types/threadActivity.ts
export type RunningThreadSummaryStatus = "scanning" | "ready" | "stale" | "unavailable";

export interface RunningThreadModelBreakdown {
  model: string | null;
  reasoningEffort: string | null;
  count: number;
}

export interface RunningThreadMember {
  threadId: string;
  title: string | null;
  model: string | null;
  reasoningEffort: string | null;
}

export interface RunningThreadGroup {
  mainThread: RunningThreadMember;
  subagents: RunningThreadMember[];
}

export interface RunningThreadSummary {
  total: number | null;
  mainThreads: number | null;
  subagents: number | null;
  mainModels: RunningThreadModelBreakdown[];
  subagentModels: RunningThreadModelBreakdown[];
  groups: RunningThreadGroup[];
  unassignedSubagents: RunningThreadMember[];
  status: RunningThreadSummaryStatus;
  updatedAt: number | null;
  detail: string;
  livenessLeaseHours: number;
}

File: tauri-app/src-tauri/src/models/thread_activity.rs
use serde::Serialize;

#[derive(Clone, Debug, PartialEq, Eq, Serialize)]
#[serde(rename_all = "camelCase")]
pub struct RunningThreadModelBreakdown {
    pub model: Option<String>,
    pub reasoning_effort: Option<String>,
    pub count: u32,
}

#[derive(Clone, Debug, PartialEq, Eq, Serialize)]
#[serde(rename_all = "camelCase")]
pub struct RunningThreadMember {
    pub thread_id: String,
    pub title: Option<String>,
    pub model: Option<String>,
    pub reasoning_effort: Option<String>,
}

#[derive(Clone, Debug, PartialEq, Eq, Serialize)]
#[serde(rename_all = "camelCase")]
pub struct RunningThreadGroup {
    pub main_thread: RunningThreadMember,
    pub subagents: Vec<RunningThreadMember>,
}

#[derive(Clone, Debug, PartialEq, Eq, Serialize)]
#[serde(rename_all = "camelCase")]
pub struct RunningThreadSummary {
    pub total: Option<u32>,
    pub main_threads: Option<u32>,
    pub subagents: Option<u32>,
    pub main_models: Vec<RunningThreadModelBreakdown>,
    pub subagent_models: Vec<RunningThreadModelBreakdown>,
    pub groups: Vec<RunningThreadGroup>,
    pub unassigned_subagents: Vec<RunningThreadMember>,
    pub status: String,
    pub updated_at: Option<i64>,
    pub detail: String,
    pub liveness_lease_hours: u32,
}
